students/douglas_klos/session4/mailroom2.py

```diff
@@ -0,0 +1,245 @@
+                    '\tThank you for your most recenet donation of ${:,.2f}.\n'
+                    '\tYour total generosity towards us is ${:,.2f}.\n'
+                    '\tIt will be put to very good use.\n'
+                    '\t\tSincerely,\n'
+                    '\t\t\tThe Team\n')
+
+
+def display_database():
+    """ Displays all donors and their donations """
+
+    print()
+
+    for key in mailroom_db:
+        print(f'{key:>24} : {mailroom_db[key]:}')
+
+
+def create_report():
+    """ Prints a report of donors and their donations """
+
+    print()
+    print("Donor Name\t\t| Total Given\t\t| Num Gifts | Average Gift")
+    print("-" * 79)
+    for key in mailroom_db:
+        if mailroom_db[key] == []:
+            print(f'{key:24s}\t\t       {len(mailroom_db[key]):10}')
+        else:
+            print(f'{key:24s} '
+                  f'$ {sum(mailroom_db[key]):16,.2f}    '
+                  f'{len(mailroom_db[key]):10}    '
+                  f'$ {sum(mailroom_db[key])/len(mailroom_db[key]):16,.2f}')
+
+
+def add_donor(name_input=''):
+    """ Adds name_input to the database """
+
+    # Check to see if a name was passed in, if not, read user input
+    while name_input in '':
+        name_input = input("Please enter new donor's name: ")
+        if name_input in mailroom_db.keys():
+            print('Already exists')
+            name_input = ''
+        if name_input in ('q', 'Q'):
+            return
+
+    mailroom_db[name_input] = []
+
+
+def add_donation():
+    """ Add new donations to a current donor """
+
+    display_database()
+    while True:
+        name = input('\nPlease select a donor to add a donation to: ')
+        if name.lower() in ('q', 'quit'):
+            return
+        if name not in mailroom_db.keys():
+            print('Donor not found, please select from list')
+        else:
+            break
+
+    while True:
+        try:
+            donation_input = float(input('Please enter a donation: '))
+            if donation_input < 0:
+                print('Donation must be a positive number')
+            else:
+                mailroom_db[name].append(donation_input)
+                break
+        except ValueError:
+            print(f'{donation_input} is not a valid entry')
+
+
+def remove_donor(name_input=''):
+    """ Remove donor from the database """
+
+    # Check to see if a name was passed in, if not, read user input
+    while name_input in '':
+        name_input = input("Please enter new donor's name: ")
+        if name_input in ('q', 'Q'):
+            return
+        if name_input in mailroom_db.keys():
+            del mailroom_db[name_input]
+        else:
+            print(f'Donor {name_input} not found')
+
+
+def thank_you_menu():
+    """ Menu for Thank You options """
+
+    while True:
+        name_input = input(THANK_YOU_PROMPT)
+        if name_input.lower() in ('q', 'quit'):
+            return
+        elif name_input.lower() in('l', 'list'):
```

Comment:
This _elif_ statement is not necessary, as it is preceded by a _return_, you can change it to an _if_.